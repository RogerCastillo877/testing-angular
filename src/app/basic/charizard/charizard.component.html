<h2>Pok√©mon</h2>
<div *ngIf="!charizard ; else elseBlock">
  <h3>Loading...</h3>
</div>

<ng-template #elseBlock>
  <h3>{{ charizard?.name | titlecase }}</h3>
  <img [src]="charizard?.sprites?.front_default" [alt]="charizard?.name">
</ng-template>