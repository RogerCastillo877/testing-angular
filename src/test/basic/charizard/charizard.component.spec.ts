import { ComponentFixture, TestBed } from '@angular/core/testing';

import { CharizardComponent } from '../../../../src/app/basic/charizard/charizard.component';
import { PokemonService } from '../../../../src/app/basic/services/pokemon.service';
import { HttpClientTestingModule, HttpTestingController } from '@angular/common/http/testing';

describe('CharizardComponent', () => {
  let component: CharizardComponent;
  let fixture: ComponentFixture<CharizardComponent>;
  let compiled: HTMLElement;
  let service: PokemonService;
  let httpMock: HttpTestingController;

  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [CharizardComponent],
      imports: [HttpClientTestingModule],
      providers: [PokemonService]
    })
      .compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(CharizardComponent);
    component = fixture.componentInstance;
    service = TestBed.inject(PokemonService);
    httpMock = TestBed.inject(HttpTestingController);

    fixture.detectChanges();
    compiled = fixture.nativeElement;
  })

  test('should create', () => {
    expect(component).toBeTruthy();
  });

  test('shoul be match with snapshot', () => {
    expect(compiled.innerHTML).toMatchSnapshot();
  })

  test('shoul display a loading at init', () => {
    const h3 = compiled.querySelector('h3');

    expect(h3?.textContent).toContain('Loading')
  })

  test('shoul charge charizard inmediatly', () => {

    const dummyPokemon = {
      name: 'charizardo',
      sprites: {
        front_default: 'https//charizard.com/sprite.png'
      }
    };

    const request = httpMock.expectOne('https://pokeapi.co/api/v2/pokemon/6');
    expect(request.request.method).toBe('GET');
    request.flush(dummyPokemon);

    fixture.detectChanges();
    const h3 = compiled.querySelector('h3');
    const img = compiled.querySelector('img');

    expect(h3?.textContent?.toLowerCase()).toContain(dummyPokemon.name.toLowerCase());
    expect(img?.src).toBe(`http://localhost/${dummyPokemon.sprites.front_default}`);
    expect(img?.alt).toBe(dummyPokemon.name);
  })
});
