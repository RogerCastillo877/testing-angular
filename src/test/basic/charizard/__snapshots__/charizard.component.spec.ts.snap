// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharizardComponent shoul be match with snapshot 1`] = `
"<h2>Pok√©mon</h2><div><h3>Loading...</h3></div><!--bindings={
  "ng-reflect-ng-if-else": "[object Object]"
}--><!--container-->"
`;
